crates/driver/src/domain/competition/solution/scoring.rs

```diff
@@ -158,17 +157,85 @@ impl Trade {
         Ok(price.in_eth(surplus.amount))
     }
 
-    /// Protocol fee is defined by fee policies attached to the order.
+    /// Protocol fees is defined by fee policies attached to the order.
     ///
     /// Denominated in SURPLUS token
-    fn protocol_fee(&self) -> Result<eth::Asset, Error> {
-        // TODO: support multiple fee policies
-        if self.policies.len() > 1 {
-            return Err(Error::MultipleFeePolicies);
+    fn protocol_fees(&self) -> Result<eth::Asset, Error> {
+        let mut amount = TokenAmount::default();
+        let mut current_trade = self.clone();
+        for (i, protocol_fee) in self
+            .fulfillment
+            .order()
+            .protocol_fees
+            .iter()
+            .rev()
+            .enumerate()
+        {
+            let fee = current_trade.protocol_fee(protocol_fee)?;
+            amount = amount
+                .checked_add(&fee)
+                .ok_or(Error::Math(Math::Overflow))?;
+            // Do not apply the last fee since it won't be used again, it minimizes the
+            // changes of negative trade fee
```

Comment:
Can you give a bit more context on why we don't use one fee here?
Also you can make this slightly nicer to read by doing `iter.().enumerate().rev()`. Then you can do `if i==0`.
Also would this exception cause problems if there is only 1 fee?

Reply:
> Then you can do if i==0.
Also would this exception cause problems if there is only 1 fee?

Not at all. It is not an exemption, and actually we can even remove it. The reason it is there is because when you calculate the last fee, it is not needed to recalculate again the `custom_prices`, because we won't use it anymore for the next fee calculation (since it was the last one), so I added that there for two reasons:
- avoid extra operation
- avoid potentially triggering an `Overflow` since we are doing operations which aren't meant to be done